src/drivers/github.js

```diff
@@ -70,18 +70,37 @@ class Github {
   }
 
   async commentCreate(opts = {}) {
-    const { report: body, commitSha } = opts;
-
-    const { url: commitUrl } = await octokit(
-      this.token,
-      this.repo
-    ).repos.createCommitComment({
-      ...ownerRepo({ uri: this.repo }),
-      body,
-      commit_sha: commitSha
-    });
-
-    return commitUrl;
+    const { report: body, commitSha, update, watermark } = opts;
+
+    const rest = octokit(this.token, this.repo);
+
+    const existing = Object.values(
+      await rest.paginate(rest.repos.listCommentsForCommit, {
+        ...ownerRepo({ uri: this.repo }),
+        commit_sha: commitSha
+      })
+    )
+      .filter((comment) => comment.body && comment.body.endsWith(watermark))
+      .sort((first, second) => first.id < second.id)
+      .pop();
+
+    if (update && existing) {
+      return (
+        await rest.repos.updateCommitComment({
+          ...ownerRepo({ uri: this.repo }),
+          comment_id: existing.id,
+          body
+        })
+      ).data.html_url;
+    } else {
+      return (
+        await rest.repos.createCommitComment({
+          ...ownerRepo({ uri: this.repo }),
+          commit_sha: commitSha,
+          body
+        })
+      ).data.html_url;
+    }
```

Comment:
@DavidGOrtega dislikes this kind of structure with several `return` statements that run exclusively from the different branches of a conditional. I find that, in this specific case, the intention is clearer than with the terse alternative, but I don't care that much if any of you prefers to write it differently.

Reply:
I would say its never clearer and adds an indentation, if its the last thing to be done

Reply:
```suggestion
    if (update && existing)
      return (
        await rest.repos.updateCommitComment({
          ...ownerRepo({ uri: this.repo }),
          comment_id: existing.id,
          body
        })
      ).data.html_url;
      
    return (
      await rest.repos.createCommitComment({
        ...ownerRepo({ uri: this.repo }),
        commit_sha: commitSha,
        body
      })
    ).data.html_url;
```

Would you prefer it like that?

Reply:
Some extensive discussions on both appreaches:

https://stackoverflow.com/questions/9191388/it-is-more-efficient-to-use-if-return-return-or-if-else-return
https://softwareengineering.stackexchange.com/questions/127128/what-is-the-best-style-for-functions-with-multiple-returns-and-if-else-statement
https://softwareengineering.stackexchange.com/questions/157407/best-practice-on-if-return

It looks like this kind of choice is the ideal breeding ground for a holy war with a copious arsenal for both bands. 😅  As stated, I don't have any strong opinion on this and I'm happy to adopt the alternate style for consistency with the project. Feel free to apply the suggestion.

Reply:
In my university this was not a holy war at all.
If you did so or did bolean === 'true' you where automatically wasted 😁